crypto/o_str.c

```diff
@@ -248,3 +249,20 @@ int openssl_strerror_r(int errnum, char *buf, size_t buflen)
     return 1;
 #endif
 }
+
+#ifdef OPENSSL_SYS_WINDOWS
+# if defined(_WIN64)
+/*
+ * cut strings to 2GB.  Only needed on Win64.
+ */
+unsigned int OPENSSL_strlen31(const char *str)
+{
+    unsigned int len = 0;
+
+    while (*str && len < 0x80000000U)
+        str++, len++;
+
+    return len & 0x7FFFFFFF;
+}
+# endif
+#endif
```

Comment:
You're making life difficult for yourself...  I'd suggest keeping the inline `_strlen31` instead.